src/test/java/com/internship/backend/controller/ReservationControllerTest.java

```diff
@@ -0,0 +1,194 @@
+package com.internship.backend.controller;
+
+import com.internship.backend.dto.ReservationDTO;
+import com.internship.backend.exceptions.*;
+import com.internship.backend.mappper.ReservationMapper;
+import com.internship.backend.model.Reservation;
+import com.internship.backend.model.TennisCourt;
+import com.internship.backend.service.ReservationService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@ActiveProfiles("test")
+class ReservationControllerTest {
+
+    @InjectMocks
+    private ReservationController reservationController;
+
+    @Mock
+    private ReservationService reservationService;
+
+    private MockMvc mockMvc;
+    private Reservation reservation;
+    private ReservationDTO reservationDTO;
+    private TennisCourt tennisCourt;
+    private List<Reservation> reservations;
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+        mockMvc = MockMvcBuilders.standaloneSetup(reservationController).build();
+
+        tennisCourt = createDefaultTennisCourt();
```

Comment:
you could call these methods only from the tests where you need the respective objects, not directly in the setup method. valid for the rest of the test classes